src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -463,13 +492,23 @@ void activity_classifier::perform_training_iteration() {
   training_data_iterator_->reset();
 }
 
-gl_sframe activity_classifier::perform_inference(data_iterator* data) const {
+
+gl_sframe
+activity_classifier::perform_inference(data_iterator *data,
+                                       std::string output_frequency) const {
   // Open a new SFrame for writing.
-  gl_sframe_writer writer({ "session_id", "preds", "num_samples" },
-                          { data->session_id_type(),
-                            flex_type_enum::VECTOR,
-                            flex_type_enum::INTEGER  },
-                          /* num_segments */ 1);
+  if (output_frequency == "per_window") {
```

Comment:
Instead of making the behavior of `perform_inference` parameterized by `output_frequency`, could we just have it always return an SFrame that suffices for either case. It can include both `num_samples` and `prediction_id` every time. The inclusion of `prediction_id` should not affect the per-row case; for the per-window case, you can just drop the `num_samples` column. (Or heck, maybe it's useful to keep it in there. What do you think?)